examples/util/fill.rs

```diff
@@ -75,6 +77,50 @@ mod platform {
         }
     }
 
+    #[allow(dead_code)]
+    pub fn fill_window_with_fn(
+        window: &dyn Window,
+        f: impl FnOnce(&mut [u32], usize, f32, u32, u32) -> Vec<[u32; 4]>,
+    ) {
+        GC.with(|gc| {
+            let size = window.surface_size();
+            let (Some(width), Some(height)) =
+                (NonZeroU32::new(size.width), NonZeroU32::new(size.height))
+            else {
+                return;
+            };
+
+            // Either get the last context used or create a new one.
+            let mut gc = gc.borrow_mut();
+            let surface =
+                gc.get_or_insert_with(|| GraphicsContext::new(window)).create_surface(window);
+
+            surface.resize(width, height).expect("Failed to resize the softbuffer surface");
+
+            // Run a function on the buffer.
+            let mut buffer = surface.buffer_mut().expect("Failed to get the softbuffer buffer");
+            let rects = f(
+                &mut buffer,
+                u32::from(width) as usize,
+                window.scale_factor() as f32,
+                width.into(),
+                height.into(),
+            );
+            // Collect only valid rects, and as softbuffer's expected type.
+            let rects = rects
+                .iter()
+                .filter(|r| r[2] != 0 && r[3] != 0)
+                .map(|r| softbuffer::Rect {
+                    x: r[0],
+                    y: r[1],
+                    width: r[2].try_into().unwrap(),
+                    height: r[3].try_into().unwrap(),
+                })
+                .collect::<Vec<_>>();
+            buffer.present_with_damage(&rects).expect("Failed to present the softbuffer buffer");
```

Comment:
It seems like you're reusing the buffer in your impl? That doesn't work unless `buffer.age() == 1` (which it e.g. isn't on macOS).

Reply:
Good catch!